lib/private/message_settings_update.php

```diff
@@ -70,9 +70,29 @@ function generateQueryStringText( $CurrentValues, &$BindValues, $ValueName, $New
     
         return '';
     }
-    
+
     // -----------------------------------------------------------------------------
-    
+
+    function getRequestValue( $aRequest, $sParameterName, $sValueType = "IntValue" )
```

Comment:
Good idea, but I would change two things:
1. Either use constants for sValueType or better still two different functions (one per type) to get rid of the if/switch altogether
2. Pass the value to change as a reference to this function. This way you can make sure that intvalues ar not assigned textvalues + you could even clear the other value.